src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -407,9 +408,27 @@ public InnReachResponseDTO finalCheckIn(String trackingId, String centralCode, B
 
     transaction.setState(FINAL_CHECKIN);
 
+    removeItemTransactionInfo(transaction.getHold().getFolioItemId())
+      .ifPresent(this::removeHoldingsTransactionInfo);
+    removeFinalCheckInFieldsFromHold(transaction);
+
     return success();
   }
 
+  private void removeFinalCheckInFieldsFromHold(InnReachTransaction transaction) {
```

Comment:
lets move this method to InnReachTransactionUtils and rename it to something more generic like `clearPatronAndItemInfo`